.github/workflows/ci.yml

```diff
@@ -180,40 +194,56 @@ jobs:
         if: matrix.name == 'Linux'
       - name: CCache stats
         run: ccache -s
+        
+  release:
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          path: artifacts
       - name: Release
-        uses: softprops/action-gh-release@v1
-        if: ${{ startsWith(github.ref, 'refs/tags/v') && matrix.release }}
+        uses: softprops/action-gh-release@v1       
         with:
-          files: pack/*
+          files: artifacts/*/*
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+  release-nightly:
+    needs: [set-nightly, build]
+    if: needs.set-nightly.outputs.nightly
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # needed for tagging
+          fetch-depth: 0
+          token: ${{ secrets.PUSH_NIGHTLY_TOKEN }}
       - name: Prepare Nightly Release
-        if: env.LEAN_VERSION_STRING
         run: |
-          # can't push shallow checkout
-          git fetch --unshallow
+          git remote add nightly https://foo:'${{ secrets.PUSH_NIGHTLY_TOKEN }}'@github.com/${{ github.repository_owner }}/lean4-nightly.git
           git fetch nightly --tags
-          if git tag $LEAN_VERSION_STRING && git push nightly $LEAN_VERSION_STRING; then
-            last_tag=$(git describe HEAD^ --abbrev=0 --tags)
-            echo -e "Changes since ${last_tag}:\n\n" > diff.md
-            #git show $last_tag:doc/changes.md > old.md
-            #./script/diff_changelogs.py old.md doc/changes.md >> diff.md
-            echo -e "*Full commit log*\n" >> diff.md
-            git log --oneline $last_tag..HEAD | sed 's/^/* /' >> diff.md
-          else
-            # make sure every runner is building the same commit
-            [ $(git rev-parse HEAD) == $(git rev-parse $LEAN_VERSION_STRING) ] || exit 11
-            echo -n > diff.md
-          fi
+          git tag ${{ needs.set-nightly.outputs.nightly }}
+          git push nightly ${{ needs.set-nightly.outputs.nightly }}
+          last_tag=$(git describe HEAD^ --abbrev=0 --tags)
+          echo -e "*Changes since ${last_tag}:*\n\n" > diff.md
+          git show $last_tag:doc/RELEASES.md > old.md
+          #./script/diff_changelogs.py old.md doc/changes.md >> diff.md
+          comm -13 old.md doc/RELEASES.md >> diff.md
```

Comment:
@leodemoura Since we talked about it last time, I realized that we did have a quite elaborate changelog diff script in Lean 3. Our new changelog has a simpler structure, so I added this very simple diff for now.